File: problem18/main.py

##82. Remove Duplicates from Sorted List II
class ListNode:
    def __init__(self, val=None, next=None):
        self.val = val
        self.next = next

    def __str__(self):
        return f"Node(value:{self.val},next:{self.next})"


class Solution:
    def deleteDuplicates(self, head: ListNode) -> ListNode:
        print(head.val==None)
        if(head.val is not None):
            list = self.getAllElement(head)
            return self.addElementWithouDuplicateValue(list)
        else:
            return ListNode()

    def getAllElement(self, head):
        newList = []
        listnode = head
        while True:
            newList.append(listnode.val)
            if listnode.next is None:
                break
            else:
                listnode = listnode.next
        return sorted(self.get_repeated_elements(newList))

    def addElementWithouDuplicateValue(self, list):
        newnode = ListNode()
        testnode = newnode
        for i in range(len(list)):
            while True:
                if testnode.next is None:
                    testnode.next = ListNode(list[i])
                    break
                else:
                    testnode = testnode.next
        return newnode.next
    def unique_elements(self,lst):
        """
        Returns a list of unique elements from the input list.
        """
        return list(set(lst))
    def get_repeated_elements(self,lst):

        unique_lst =self.unique_elements(lst)
        repeated_elements = []
        for element in unique_lst:
            if lst.count(element) == 1:
                repeated_elements.append(element)
        return repeated_elements


if __name__ == '__main__':
    l1 = ListNode(
        2, ListNode(4, ListNode(3, ListNode(5, ListNode(6, ListNode(4))))))
    l3=ListNode()
    print(Solution().deleteDuplicates(l3))

File: problem18/main.dart
//82. Remove Duplicates from Sorted List II

class ListNode {
  int val;
  var next;
  ListNode([this.val = 0, this.next = null]);
  @override
  String toString() => "Node(value:$val,next:$next)";
}

class Solution {
  ListNode? deleteDuplicates(ListNode? head) {
    var list = getAllElement(head);

    return addElementWithouDuplicateValue(list);
  }

  List<int> getAllElement(head) {
    List<int> newList = [];
    var listnode = head;
    while (true) {
      newList.add(listnode.val);
      if (listnode.next == null) {
        break;
      } else {
        listnode = listnode.next;
      }
    }

    return getUnRepeatedElements(newList);
  }

  ListNode? addElementWithouDuplicateValue(List<int> list) {
    list.sort();
    ListNode? newnode = ListNode();
    var testnode = newnode;

    for (int i = 0; i < list.length; i++) {
      while (true) {
        if (testnode.next == null) {
          testnode.next = ListNode(list[i]);
          break;
        } else {
          testnode = testnode.next;
        }
      }
    }
    return newnode.next;
  }

  List<int> getUnRepeatedElements(List lst) {
    var uniqueLst = lst.toSet().toList();
    var repeatedElements = <int>[];
    for (var element in uniqueLst) {
      if (lst.where((e) => e == element).length == 1) {
        repeatedElements.add(element);
      }
    }
    return repeatedElements;
  }
}

void main(List<String> args) {
  ListNode l1 = ListNode(
      1, ListNode(1, ListNode(1, ListNode(2, ListNode(3, ListNode(4))))));
  print(Solution().deleteDuplicates(l1));
}
